### PR TITLE
Color-code asset-types on map

### PR DESCRIPTION
**Pull request by Arth101**
Toggl: `Indiker asset-typer på kortvisning med ikoner og måske farver?`

I had to create a new helper function, because chaning `determineMediaType` would interfere with the player type on the asset item page.

image
image

### CODE REVIEW

File: assets-pipeline/app/scripts/historisk-atlas.js [hunk 1]
```diff
@@ -1143,10 +1143,34 @@ function HistoriskAtlas(mapElement, options) {
...
+
+      if(selected) {
+        return options.icons.assetSelected;
+      }
+
+      return options.icons.asset;
+    }
+
+    if(mapState.isEditMode()) {
+      return options.icons.assetHeadingEdit
```
Comment (hypesystem) [inline assets-pipeline/app/scripts/historisk-atlas.js:1160]: Missing ;

File: assets-pipeline/app/scripts/historisk-atlas.js [hunk 2]
```diff
@@ -1143,10 +1143,34 @@ function HistoriskAtlas(mapElement, options) {
...
+    }
+
+    if(selected) {
+      return options.icons.assetHeadingSelected
```
Comment (hypesystem) [inline assets-pipeline/app/scripts/historisk-atlas.js:1164]: Missing ;

File: assets-pipeline/app/scripts-browserify/map-controller.js [hunk 3]
```diff
@@ -29,6 +31,7 @@ function _mapEsResultsToAssets(results, searchParameters) {
     .map((hit) => {
       const asset = hit._source;
       const colid = `${asset.collection}/${asset.id}`;
+      console.warn("RAW asset", asset);
```
Comment (hypesystem) [inline assets-pipeline/app/scripts-browserify/map-controller.js:34]: Fine for debugging, should be removed

**Comment by hypesystem** [comment]
Why draft?

File: assets-pipeline/app/scripts-browserify/map-controller.js [hunk 4]
```diff
@@ -135,9 +135,7 @@ function MapController (mapElement, searchControllerCallbacks, options) {
       // different set for single assets.
       if (options.mode === 'single') {
         options.icons.asset = options.icons.assetSelected;
-        options.icons.assetHeading = '../images/icons/map/pinheadingselected.png';
-        options.icons.assetEdit = '../images/icons/map/pinedit.png';
-        options.icons.assetHeadingEdit = '../images/icons/map/pinheadingedit.png';
+        options.icons.assetHeading = options.icons.assetHeadingSelected;
```
Comment (hypesystem) [inline assets-pipeline/app/scripts-browserify/map-controller.js:138]: Why is `Edit` and `HeadingEdit` not passed here? I do think these can appear in single mode, right?

File: shared/helpers.js [hunk 5]
```diff
@@ -296,6 +296,34 @@ helpers.determineMediaTypes = metadata => {
...
+  const type = metadata.original_material.displaystring;
+
+  if(type === 'Lyd') {
+    return 'audio';
+  }
+  if(type === 'Film') {
+    return 'video';
+  }
+  if(type === 'Arkæologisk artikel') {
+    return '3D';
```
Comment (hypesystem) [inline shared/helpers.js:315]: For consistency I think `3d` (lowercase) would be better here?

**Review by hypesystem — COMMENTED** [review]
I *think* support is missing on the map rendered in the series view, e.g. https://kbhbilleder.dk/stadsarkitektens-samling (select "Kort" in the result view).

This searching is controlled in scripts-browserify/search/series.js

It's a close copy of scripts-browserify/search/index.js but for the series view specifically. Really we should extract all the common functionality but that has not yet happened. Could be part of this project if there's time...

Generally looks good, had a question and some notes.

File: assets-pipeline/app/scripts-browserify/map-controller.js (hunk 4 above)
Comment (Arth101) [reply assets-pipeline/app/scripts-browserify/map-controller.js:138]: Som jeg forstår koden her så erstatter den 2 ikoner, asset og assetHeading med de "selected" versioner, da det er dem der skal bruges på Asset-sidens kort (hvor der kun er et punkt på kortet.
De 2 linjer jeg har slettet satte ikonerne til at være de samme som allerede var sat i `options.icons` objektet over.

Måske har jeg misforstået den del, men det var derfor jeg slettede linjerne, da de ikke ændrede noget.

File: shared/helpers.js (hunk 5 above)
Comment (Arth101) [reply shared/helpers.js:315]: Maybe `article` would be better? Otherwise lowercase it is :)

**Comment by Arth101** [comment]
> I _think_ support is missing on the map rendered in the series view, e.g. https://kbhbilleder.dk/stadsarkitektens-samling (select "Kort" in the result view).

Good catch. I have updated and tested the code on series maps now :)

**Comment by hypesystem** [comment]
IT WOULD SEEM

YOU ARE

IN

COMPLIANCE

**Review by hypesystem — APPROVED** [review]

**Pull request merged**